src/python/pants/engine/target_test.py

```diff
@@ -262,7 +262,7 @@ class CustomField(BoolField):
         alias = "custom_field"
         default = False
 
-    union_membership = UnionMembership({FortranTarget.PluginField: [CustomField]})
+    union_membership = UnionMembership({FortranTarget._anonymous_plugin_field_cls: [CustomField]})
```

Comment:
This would be less awkward if UnionMembership had `UnionMembership.from_rules(Iterable[UnionRule) -> UnionMembership` which it arguably should have already anyhow.

Reply:
Good call. I'll do that as a precursor PR.